controllers/function_mesh.go

```diff
@@ -236,6 +236,9 @@ func (r *FunctionMeshReconciler) UpdateFunctionMesh(ctx context.Context, req ctr
 
 	for _, functionSpec := range mesh.Spec.Functions {
 		condition := mesh.Status.FunctionConditions[functionSpec.Name]
+		if functionSpec.MaxReplicas != nil && condition.Status == metav1.ConditionTrue && condition.Action == v1alpha1.NoAction {
+			continue
+		}
```

Comment:
@armangurkan  as mentioned, I added this logic to skip the reconciliation for function/sink/source that already controlled by the HPA